### PR TITLE
ENH: add option to remove bracket from mapclassify legend

### PR DESCRIPTION
**Pull request by donlo**
Address #1518 

and update choro legend example with bracket removal option

### CODE REVIEW

File: geopandas/plotting.py [hunk 1]
```diff
@@ -526,6 +526,9 @@ def plot_dataframe(
             A list of legend labels to override the auto-generated labels.
             Needs to have the same number of elements as the number of
             classes (`k`).
+        interval : boolean
+            An option to remove brackets from mapclassify legend.
+            If True, open/closed interval brackets are shown in legend.
```
Comment (martinfleis) [inline geopandas/plotting.py:531]:
```suggestion
        interval : boolean (default True)
            An option to control brackets from mapclassify legend.
            If True, open/closed interval brackets are shown in the legend.
```

File: geopandas/plotting.py [hunk 2]
```diff
@@ -693,7 +696,16 @@ def plot_dataframe(
...
+
             categories = binning.get_legend_classes(fmt)
+            show_interval = True
+            if legend_kwds is not None and "interval" in legend_kwds:
+                show_interval = legend_kwds.pop("interval")
+                if not show_interval:
+                    categories = [
+                        c.replace("(", "").replace(")", "").replace("[", "").replace("]", "")
+                        for c in categories
+                    ]
```
Comment (martinfleis) [inline geopandas/plotting.py:708]:
```suggestion
                    categories = [c[1:-1] for c in categories]
```

This would be easier.

**Review by martinfleis — COMMENTED** [review]
Thank you! I've left a few comments in the code.

Can you also make sure that the code follows `black` (latest version 20) style? See https://geopandas.readthedocs.io/en/latest/community/contributing.html#style-guide-linting

**Comment by donlo** [comment]
Thanks for the review! Made the suggested changes

**Review by martinfleis — APPROVED** [review]
Thanks!

**Comment by jorisvandenbossche** [comment]
This has conflicts now (with the changed examples). 

One thing I am wondering is if we shouldn't make `interval=False` the default? 
(it's basically what we had before I think (before mapclassify changed it), and personally, I think it might be what more people might want to have)

**Comment by martinfleis** [comment]
> One thing I am wondering is if we shouldn't make interval=False the default?

That is actually quite a good point. 

I'll fix the conflict we made and change the default.

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Thanks @donlo (sorry for the slow follow-up, but it will be included in the 0.9 release) and @martinfleis !

**Pull request merged**